js/packages/teams-dialogs/src/prompts/teamsBotSsoPrompt.ts

```diff
@@ -0,0 +1,470 @@
+ * 
+ * @remarks
+ * The prompt will attempt to retrieve the users current token of the desired scopes and store it in
+ * the token store.
+ * 
+ * User will be automatically signed in leveraging Teams support of Bot Single Sign On(SSO):
+ * https://docs.microsoft.com/en-us/microsoftteams/platform/bots/how-to/authentication/auth-aad-sso-bots
+ * 
+ * > [!NOTE]
+ * > You should avoid persisting the access token with your bots other state. The Bot Frameworks
+ * > SSO service will securely store the token on your behalf. If you store it in your bots state
+ * > it could expire or be revoked in between turns.
+ * >
+ * > When calling the prompt from within a waterfall step you should use the token within the step
+ * > following the prompt and then let the token go out of scope at the end of your function.
+ * 
+ * @example
+ * When used with your bots `DialogSet` you can simply add a new instance of the prompt as a named
+ * dialog using `DialogSet.add()`. You can then start the prompt from a waterfall step using either
+ * `DialogContext.beginDialog()` or `DialogContext.prompt()`. The user will be prompted to sign in as
+ * needed and their access token will be passed as an argument to the callers next waterfall step:
+ *
+ * ```JavaScript
+ * const { ConversationState, MemoryStorage } = require('botbuilder');
+ * const { DialogSet, WaterfallDialog, TeamsBotSsoPrompt } = require('botbuilder-dialogs');
+ *
+ * const convoState = new ConversationState(new MemoryStorage());
+ * const dialogState = convoState.createProperty('dialogState');
+ * const dialogs = new DialogSet(dialogState);
+ *
+ * dialogs.add(new TeamsBotSsoPrompt('TeamsBotSsoPrompt', {
+ *    scopes: ["User.Read"],
+ * }));
+ *
+ * dialogs.add(new WaterfallDialog('taskNeedingLogin', [
+ *      async (step) => {
+ *          return await step.beginDialog('TeamsBotSsoPrompt');
+ *      },
+ *      async (step) => {
+ *          const token = step.result;
+ *          if (token) {
+ *
+ *              // ... continue with task needing access token ...
+ *
+ *          } else {
+ *              await step.context.sendActivity(`Sorry... We couldn't log you in. Try again later.`);
+ *              return await step.endDialog();
+ *          }
+ *      }
+ * ]));
+ * ```
+ */
+export class TeamsBotSsoPrompt extends Dialog {
+  private initiateLoginEndpoint: string;
+  private settings: TeamsBotSsoPromptSettings;
+  private authConfig: TeamsBotSsoPromptAuthConfig;
+
+  constructor(
+    authConfig: TeamsBotSsoPromptAuthConfig,
+    initiateLoginEndpoint: string,
+    dialogId: string,
+    settings: TeamsBotSsoPromptSettings,
+  ) {
+    super(dialogId);
+    this.authConfig = authConfig;
+    this.settings = settings;
+    this.initiateLoginEndpoint = initiateLoginEndpoint;
+
+    this.validateScopesType(settings.scopes);
+  }
+
+  /**
+   * Called when a prompt dialog is pushed onto the dialog stack and is being activated.
+   * @remarks
```

Comment:
Please change `@remarks` to `@summary` :) (remarks is deprecated and I am trying to clean up the usage of it)